File: solutions/javascript/blind-75/tree/maximumDepthOfBinaryTree.js
const maxDepth = (root) => {
    if (!root) return 0;
    
    const leftSubtreeDepth = maxDepth(root.left);
    const rightSubtreeDepth = maxDepth(root.right);
    
    return Math.max(leftSubtreeDepth, rightSubtreeDepth) + 1;
};

File: solutions/python/blind-75/tree/maximum_depth_of_binary_tree.py

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        left_subtree_depth = self.maxDepth(root.left)
        right_subtree_depth = self.maxDepth(root.right)
        
        return max(left_subtree_depth, right_subtree_depth) + 1